selector/diversity.py

```diff
@@ -375,15 +375,25 @@ def hypersphere_overlap_of_subset(x: np.ndarray, x_subset: np.array) -> float:
     Agrafiotis, D. K.. (1997) Stochastic Algorithms for Maximizing Molecular Diversity.
     Journal of Chemical Information and Computer Sciences 37, 841-851.
     """
-    d = len(x_subset[0])
-    k = len(x_subset[:, 0])
+
     # Find the maximum and minimum over each feature across all molecules.
     max_x = np.max(x, axis=0)
     min_x = np.min(x, axis=0)
-    # normalization of each feature to [0, 1]
-    if np.any(np.abs(max_x - min_x) < 1e-30):
-        raise ValueError("One of the features is redundant and causes normalization to fail.")
 
+    if np.all(np.abs(max_x - min_x) < 1e-30):
+        raise ValueError("All of the features are redundant which causes normalization to fail.")
```

Comment:
Should this be a warning instead of a `ValueError`? Otherwise, the function will break and the following code will not work. @marco-2023

Reply:
@FanwangM In this case it is appropriate to raise an error. The if statement here checks, for all columns, if the max and min values differ by less than `1e-30`. This is equivalent to all elements in each column being the same (all the samples in the data are equal).